modules/tinymce/src/plugins/lists/main/ts/actions/ToggleList.ts

```diff
@@ -191,10 +191,19 @@ const hasCompatibleStyle = (dom: DOMUtils, sib: Element, detail: ListDetail): bo
   return sibStyle === detailStyle;
 };
 
+const getRootSearchStart = (editor: Editor, range: Range): Node => {
```

Comment:
This seems to not work for cases with `<div><b>f[oo</b></div><div><b>bar</b></div><div><b>b]az</b></div>` so it should work with formatted nodes as well not just plain text nodes.

Reply:
Checking for forced_root_block is not a good idea, since it only acts as an easy trigger, but isn't actually the cause of the issue. The cause is the divs themselves, whether they  come from force_root_block, setContent, or anything else. Just checking for the body is also not a good idea, since this issue can occur in other places too, such as nested divs. It'd be easier, as you point out, but I don't think it'd be a good solution. I'll look into our code for detecting the block to move into an li and use that for how to step above it.